Automattic/wp-calypso
# Removed access-control.js from plugins to remove sites-list

This pull request removes access-control from plugins and so is one less sites-list usage and big a removal.
This PR is dependent on PR https://github.com/Automattic/wp-calypso/pull/16248 and https://github.com/Automattic/wp-calypso/pull/16247. That create new components used here.

Besides removing sites-list the pull request also creates a separation of concerns for access-control in plugins.

In order to remove access-control I had to change all the places that use access-control (so this pull request is error prone):
•	client/my-sites/plugins/plugins-browser
•	client/my-sites/plugins/plugin.jsx
•	client/my-sites/plugins/main.jsx

**To test:**
Test the following url combinations:
•	http://calypso.localhost:3000/plugins/{site/no-site/jetpack-site}, --
•	http://calypso.localhost:3000/plugins/hello-dolly/{site/no-site/jetpack-site},
•	http://calypso.localhost:3000/plugins/browse/{site/no-site/jetpack-site}
Test on users with permission to manage plugins and on users without permission to manage plugins (make sure in the latter case not found pages appear).

Use the plugins section and make sure everything is behaving as it should and no new not reported bugs were created.
